### PR TITLE
*: 'go test --short ./...' runs all UNIT tests in <25s

### PR DESCRIPTION
**Pull request by ptabor**
Marked all integration+e2e tests as skipped in the --short mode.

Thanks to this we will be able to significantly simplify ./test script in follow up CL 
[needed for modularization].

The run currently takes ~23s.
With (follow up) move of ~clientv3/snapshot to integration tests (as
part of modularization), we can expect this to fall to 5-10s.

```
% time go test --short ./... --count=1
ok  	go.etcd.io/etcd/v3	0.098s
ok  	go.etcd.io/etcd/v3/auth	0.724s
...
(slowest:)
ok  	go.etcd.io/etcd/v3/clientv3	3.219s
ok  	go.etcd.io/etcd/v3/clientv3/balancer	1.102s
ok  	go.etcd.io/etcd/v3/clientv3/concurrency	3.323s
ok  	go.etcd.io/etcd/v3/clientv3/snapshot	19.325s
ok  	go.etcd.io/etcd/v3/contrib/raftexample	7.572s
ok  	go.etcd.io/etcd/v3/etcdserver	1.698s
ok  	go.etcd.io/etcd/v3/etcdserver/api/v3compactor	1.765s
ok  	go.etcd.io/etcd/v3/lease	3.455s
ok  	go.etcd.io/etcd/v3/lease/leasehttp	2.185s
ok  	go.etcd.io/etcd/v3/mvcc	7.246s
ok  	go.etcd.io/etcd/v3/pkg/netutil	1.024s
ok  	go.etcd.io/etcd/v3/pkg/proxy	4.081s
ok  	go.etcd.io/etcd/v3/pkg/tlsutil	3.965s
ok  	go.etcd.io/etcd/v3/wal	1.517s
...
go test --short ./... --count=1  76.12s user 12.57s system 375% cpu 23.635 total
```

### CODE REVIEW

File: etcdserver/api/v3compactor/periodic.go [hunk 1]
```diff
@@ -136,13 +136,14 @@ func (pc *Periodic) Run() {
 				zap.Int64("revision", rev),
 				zap.Duration("compact-period", pc.period),
 			)
+			startTime := pc.clock.Now()
```
Comment (jingyih) [inline etcdserver/api/v3compactor/periodic.go:139]: Could we fix this in a separate PR?

File: tests/e2e/ctl_v2_test.go [hunk 2]
```diff
@@ -30,6 +30,8 @@ func TestCtlV2SetClientTLS(t *testing.T) { testCtlV2Set(t, &configClientTLS, fal
 func TestCtlV2SetPeerTLS(t *testing.T)   { testCtlV2Set(t, &configPeerTLS, false) }
 func TestCtlV2SetTLS(t *testing.T)       { testCtlV2Set(t, &configTLS, false) }
 func testCtlV2Set(t *testing.T, cfg *etcdProcessClusterConfig, quorum bool) {
+	skipInShortMode(t)
```
Comment (jingyih) [inline tests/e2e/ctl_v2_test.go:33]: For the purpose of skipping e2e tests, is it sufficient to add `skipInShortMode(t)` to https://github.com/etcd-io/etcd/blob/01bd0d468d3eebd2825390fc7bcd17a5ec57d389/tests/e2e/cluster_test.go#L144:6? If so, the code is cleaner.

File: etcdserver/api/v3compactor/periodic.go (hunk 1 above)
Comment (ptabor) [reply etcdserver/api/v3compactor/periodic.go]: Done. 

https://github.com/etcd-io/etcd/pull/12303

File: tests/e2e/ctl_v2_test.go (hunk 2 above)
Comment (ptabor) [reply tests/e2e/ctl_v2_test.go:33]: This method does not have access to testing.T (nor testing.TB). 

If you prefer I can add the param (t) to the O(10) invocations of newEtcdProcessCluster(t, ...)

Comment (jingyih) [reply tests/e2e/ctl_v2_test.go:33]: Oh I see, so it is comparable lines of code changes. However I still prefer to pass testing.TB to newEtcdProcessCluster() due to better code readability and maintainability. If later some other developer contribute a new e2e test, it is guaranteed to be skipped in short mode. What do you think?

Comment (ptabor) [reply tests/e2e/ctl_v2_test.go:33]: SGTM. On it.

**Comment by ptabor** [comment]
@jingyih comments addressed. PTAL

**Review by jingyih — COMMENTED** [review]
lgtm
Thanks!

**Pull request merged**